designs/build_debug_artifacts.md

```diff
@@ -0,0 +1,120 @@
+SAM Build producing debuggable artifacts
+========================================
+
+What is the problem?
+--------------------
+
+`sam build` produces artifacts that are built for production use. In some langauges (usually interpreted), 
+these production artifacts are also debuggable locally. In the case of compiled languages, you usually need to compile
+the binary or artifact in a specific manner for them to be debuggable.  
+
+What will be changed?
+---------------------
+
+We will introduce a way in `sam build` to produce these debuggable artifacts for those compiled languages.
+
+Success criteria for the change
+-------------------------------
+
+1. Artifacts generated will be debuggable for DotNetCore2.1 Runtimes
+
+Out-of-Scope
+------------
+
+1. Other languages `sam build` supports will not be changed
+
+User Experience Walkthrough
+---------------------------
+
+Implementation
+==============
+
+CLI Changes
+-----------
+
+*Explain the changes to command line interface, including adding new
+commands, modifying arguments etc*
+
+### Breaking Change
+
+Changes are additive and will not break any existing functionality.
+
+Design
+------
+
+Options considered
+------------------
+We have a couple options to consider:
+
+1. A command line option (`--debug-artifact`)
+    * Pros
+        * Customer can control what kind of artifacts are produced.
+        * There is no guessing needed by the CLI
+    * Cons
+        * Yet another option for customers to learn or know about
+        * Makes the customer need to think about the artifacts they need to produce (though this can be hidden
+        behind the AWS Toolkit in IDEs)
+        * Customers running in the command line need to remember what artifacts to produce or what they previously produced
```

Comment:
this is the case, even if you using are the native binary to do it. eg: dotnet publish -c Debug. although I'm not sure what artifacts are produced if no configurations are specified.

Reply:
For DotNet, I believe it is Release by default and same with Go (since you have to add the compile flag at build time). 

I did consider following something like DotNet here. That is a `--artifacts Debug` but I don't see another build case, so flag seemed better to me.